sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2392,6 +2399,13 @@ object functions {
   def minute(columnName: String): Column = minute(Column(columnName))
 
   /**
+   * Returns the first date which is later than given date sd and named as dow.
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def next_day(sd: Column, dow: Column): Column = NextDay(sd.expr, dow.expr)
```

Comment:
also update the scaladoc to give an example, and explain what the acceptable values are for dayOfWeek